cmd/probe/smartprobe.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"github.com/golang/glog"
+	"github.com/openebs/node-disk-manager/cmd/controller"
+	"github.com/openebs/node-disk-manager/pkg/smart"
+)
+
+// smartProbe contains required variables for populating diskInfo
+type smartProbe struct {
+	Controller      *controller.Controller
```

Comment:
comments on what this controller is all about. how is a controller associated with a probe. why should a controller be associated with a probe. is this a k8s controller. why is a probe coupled to controller.

Above were some of my questions. Ignore if they are not relevant. However, it is important to provide code level comments that explains the design.

Reply:
I will provide the comments accordingly @AmitKumarDas